### PR TITLE
Relocate storage and cover source-mode environment variable

### PR DESCRIPTION
**Pull request by GabrielDuf**
This pull request introduces a new mechanism for controlling the storage location and source mode in Pinget via environment variables, and updates the version to 0.11.0 across all components. It adds robust logic for parsing and resolving the source mode from environment variables, updates the documentation, and expands test coverage to ensure correct behavior. These changes improve flexibility for embedding, sandboxing, and customizing Pinget's behavior in different environments.

**Environment Variable Support and Source Mode Logic:**

* Added support for two environment variables: `PINGET_APPROOT` (controls storage root) and `PINGET_SOURCE_MODE` (controls source mode: `auto`, `private`, or `system-winget-mirror`) in both the Rust CLI and C# core. The logic ensures that explicit options override environment variables, and sensible defaults are chosen when variables are unset. (`README.md`, `rust/crates/pinget-core/src/lib.rs`, `dotnet/src/Devolutions.Pinget.Core/Repository.cs`) 
* Implemented parsing and resolution methods for source mode, handling various spellings and prioritization between explicit options and environment variables. (`rust/crates/pinget-core/src/lib.rs`, `dotnet/src/Devolutions.Pinget.Core/Repository.cs`) 

**Documentation and User Guidance:**

* Added detailed documentation in `README.md` explaining the new environment variable behavior, including fallback logic and embedding scenarios.

**Test Coverage Improvements:**

* Added comprehensive unit tests for the new source mode parsing and resolution logic, including edge cases and environment variable interactions. (`dotnet/src/Devolutions.Pinget.Core.Tests/CoreTests.cs`)

**Version Bump and Consistency:**

* Updated version numbers to 0.11.0 across all Rust crates, .NET projects, and PowerShell modules to reflect the new features. (`dotnet/src/Devolutions.Pinget.Cli/Program.cs`, `dotnet/src/Devolutions.Pinget.PowerShell.Cmdlets/ModuleFiles/Devolutions.Pinget.Client.psd1`, `dotnet/src/Devolutions.Pinget.PowerShell.Engine/PowerShellEngineVersion.cs`, `nuget/Devolutions.Pinget.Cli.DotNet/Devolutions.Pinget.Cli.DotNet.csproj`, `nuget/Devolutions.Pinget.Cli.Rust/Devolutions.Pinget.Cli.Rust.csproj`, `rust/crates/pinget-cli/Cargo.toml`, `rust/crates/pinget-com/Cargo.toml`, `rust/crates/pinget-core/Cargo.toml`) 

**Test Infrastructure:**

* Added `[CollectionDefinition]` attributes to test classes to ensure proper test isolation for repository state. (`dotnet/src/Devolutions.Pinget.Core.Tests/CoreTests.cs`)

### CODE REVIEW

File: dotnet/src/Devolutions.Pinget.Core/Repository.cs [hunk 1]
```diff
@@ -79,6 +78,46 @@ public static Repository Open(RepositoryOptions? options = null)
...
+    internal static SourceMode ResolveRequestedSourceMode(
+        SourceMode configuredSourceMode,
+        string? requestedAppRoot,
+        string? sourceModeFromEnvironment)
+    {
+        if (configuredSourceMode != SourceMode.Auto)
+            return configuredSourceMode;
+
+        if (TryParseSourceMode(sourceModeFromEnvironment, out var parsedSourceMode))
+            return parsedSourceMode;
```
[the remark this answers is not in the document]
Comment (GabrielDuf) [reply dotnet/src/Devolutions.Pinget.Core/Repository.cs]: [bd8f440](https://github.com/Devolutions/pinget/pull/42/commits/bd8f440e7a4d072f419b80fdfad66cbd275743d0)

File: rust/crates/pinget-core/src/lib.rs [hunk 2]
```diff
@@ -11147,21 +11168,82 @@ mod tests {
...
+        assert_eq!(SourceMode::parse("system_winget_mirror"), None);
+    }
+
+    #[test]
+    fn source_mode_reads_its_environment_variable() {
+        // The literal is deliberate: reading SOURCE_MODE_ENV_VAR here would keep this test
+        // self-consistent with a renamed constant instead of catching the rename.
+        let prior = std::env::var_os("PINGET_SOURCE_MODE");
+        // SAFETY: no other test in this module reads or writes PINGET_SOURCE_MODE.
+        unsafe { std::env::set_var("PINGET_SOURCE_MODE", "system-winget-mirror") };
```
[the remark this answers is not in the document]
Comment (GabrielDuf) [reply rust/crates/pinget-core/src/lib.rs]: [bd8f440](https://github.com/Devolutions/pinget/pull/42/commits/bd8f440e7a4d072f419b80fdfad66cbd275743d0)

**Review by mamoreau-devolutions — APPROVED** [review]
